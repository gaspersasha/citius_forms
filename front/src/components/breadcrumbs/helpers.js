export const breadcrumbTitles = {
  '/cars': 'Cars for sale',
  '/cars/new-cars': 'New Cars in Stock',
  '/cars/part-exchange': 'Get A Quote For Your Car',
  '/vans': 'Vans For Sale',
  '/leasing': 'Cars To Lease',
  '/motoring-services': 'Motoring Services',
  '/how-it-works': 'How It Works',
  '/how-it-works/finance-options': 'Finance Options',
  '/how-it-works/warranty-insurance': 'Warranty & Insurance',
  '/how-it-works/part-exchange': 'Part Exchange',
  '/how-it-works/cancellation-returns': 'Cancellations & Returns',
  '/how-it-works/faq': 'FAQ',
  '/how-it-works/about': 'About Us',
  '/how-it-works/contact-us': 'Contact us',
  '/account': 'My Account',
  '/account/my-cars': 'My Cars',
  '/account/details': 'My Details',
  '/account/orders': 'My Orders',
  // TODO: Temporary duplicated for hidden routes
  '/account/my-details': 'My Details',
  '/account/my-orders': 'My Orders',
  '/part-exchange': 'Your Current Cars',
  '/finance-thank-you': 'Thank You',
  // ...others
};
