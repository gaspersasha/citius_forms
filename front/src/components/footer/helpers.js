import { CONTACTS, URL } from '~constants';

const { FACEBOOK, TWITTER, LINKEDIN, INSTAGRAM, PRIVACY, AUTOVIA_ORIGIN } = URL;

export const links = [
  { title: 'Reservation', href: '/cars' },
  { title: 'Used Cars', href: '/cars' },
  { title: 'New Cars in Stock', href: '/cars/new-cars' },
  { title: 'Vans', href: '/vans' },
  { title: 'How It Works', href: '/how-it-works' },
  { title: 'FAQ', href: '/how-it-works/faq' },
  { title: 'About Us', href: '/how-it-works/about' },
  { title: 'As seen in the press', href: '/press' },
  { title: "Buyer's Guides", href: '/buyers-guides' },
  { title: 'News', href: '/news' },
  { title: 'Advice', href: '/advice' },
  { title: 'Complaints', href: '/how-it-works/contact-us#complaints' },
  {
    title: 'Initial Disclosure Document',
    href: '/cars/1224/initial-disclosure-document',
  },
  {
    title: 'Work For Us',
    href: `${AUTOVIA_ORIGIN}/careers/search/?department=&brand=BuyaCar&location=&typeOfEmployment=`,
    target: '_blank',
  },
  { title: 'Dealer Sales', href: '/dealer-sales' },
  {
    title: 'Privacy Policy',
    href: PRIVACY,
    target: '_blank',
  },
  {
    title: 'Cookie Policy',
    href: `${AUTOVIA_ORIGIN}/cookies-policy`,
    target: '_blank',
  },
  { title: 'Sitemap', href: '/sitemap' },
];

export const footerSocials = [
  {
    title: 'facebook',
    href: FACEBOOK,
    target: '_blank',
    rel: 'noreferrer',
  },
  {
    title: 'twitter',
    href: TWITTER,
    target: '_blank',
    rel: 'noreferrer',
  },
  {
    title: 'linkedin',
    href: LINKEDIN,
    target: '_blank',
    rel: 'noreferrer',
  },
  {
    title: 'email',
    href: `mailto:${CONTACTS.EMAIL}`,
  },
  {
    title: 'instagram',
    href: INSTAGRAM,
    target: '_blank',
    rel: 'noreferrer',
  },
];
