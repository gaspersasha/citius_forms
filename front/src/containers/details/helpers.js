import { VALIDATION_TYPES } from '~constants';
import { getFormFiledProperties, composeFormState } from '~utils';

export const composeDetailsState = (state) => {
  const additional = {
    emailConfirm: getFormFiledProperties(state.email, VALIDATION_TYPES.EMAIL),
    password: getFormFiledProperties('', VALIDATION_TYPES.PASSWORD),
    passwordConfirm: getFormFiledProperties('', VALIDATION_TYPES.PASSWORD),
  };

  return composeFormState(state, additional);
};
