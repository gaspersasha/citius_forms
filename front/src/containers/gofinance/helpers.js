import { capitalize, GTM_EVENT, GTM as gtm } from '~utils';

const GTM_MAP = {
  start: 'ssFinanceApplicationStart',
  drivingLicenceType: 'ssDrivingLicense',
  maritalStatus: 'ssMaritalStatus',
  validUKPassport: 'ssPassport',
  dependants: 'ssDependents',
  residentialStatus: 'ssResidentialStatus',
  addressConfirm: 'ssAddressConfirmation',
  addressChange: 'ssAddressChange',
  residenceYears: 'ssResidedLength',
  employmentType: 'ssEmploymentType',
  employer: 'ssEmployer',
  postCode: 'ssEmployerAddress',
  employmentYears: 'ssEmploymentLength',
  annualSalary: 'ssAnnualIncome',
  accountNumber: 'ssBankDetails',
  yearsAtBank: 'ssBankLength',
};

// Self Service Journey
export const sendGTM = (some) => {
  GTM_EVENT(GTM_MAP[some]);
};

export const handleGTM = (state, keys) => {
  Object.keys(state).forEach((key) => {
    const pos = keys.indexOf(key);

    if (pos > -1) sendGTM(keys[pos]);
  });
};

export const GTM = (path) => gtm(path, 'finance', 'Finance your car');

export const formatOption = (type = '') => {
  const word = type.split('_').join(' ').toLowerCase();

  return capitalize(word);
};

export const getYearsAtJobLabel = (status, type) => {
  if (status !== 'unemployed')
    return 'How many years have you been working at this address?';

  const option = formatOption(type).toLowerCase();

  switch (type) {
    case 'UNEMPLOYED':
    case 'RETIRED':
    case 'UNABLE_TO_WORK':
      return `How long have you been ${option}?`;
    case 'STUDENT':
    case 'HOMEMAKER':
      return `How long have you been a ${option}?`;
    default:
      return `How many years have you spent as a ${option}?`;
  }
};

export const truncateData = ({
  addresses,
  details,
  employments,
  bankDetails,
  ...rest
}) => rest;
