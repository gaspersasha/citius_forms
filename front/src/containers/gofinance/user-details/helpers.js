export const getCreditScoreLevel = (score) => {
  if (!score || typeof score !== 'number') return null;
  if (score < 50) return 'Bad';
  if (score < 239) return 'Very poor';
  if (score < 320) return 'Poor';
  if (score < 410) return 'Fair';
  if (score < 470) return 'Good';

  return 'Excellent';
};
