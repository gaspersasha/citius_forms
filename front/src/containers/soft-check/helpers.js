import {
  CarCity,
  CarConvertible,
  CarCoupe,
  CarEstate,
  CarHatchback,
  CarMPV,
  CarSaloon,
  CarSUV,
} from '~assets/svg';

export const bodyStylesData = [
  {
    name: 'Hatchback',
    link: '/cars/family-cars/hatchbacks',
    icon: CarHatchback,
  },
  {
    name: 'Estate',
    link: '/cars/family-cars/estate-cars',
    icon: CarEstate,
  },
  {
    name: 'SUV',
    link: '/cars/family-cars/suvs',
    icon: CarSUV,
  },
  {
    name: 'Saloon',
    link: '/cars/family-cars/saloon-cars',
    icon: CarSaloon,
  },
  {
    name: 'Coupe',
    link: '/cars/sports-cars/coupes',
    icon: CarCoupe,
  },
  {
    name: 'Convertible',
    link: '/cars/sports-cars/convertibles',
    icon: CarConvertible,
  },
  {
    name: 'MPV',
    link: '/cars/family-cars/mpvs',
    icon: CarMPV,
  },
  {
    name: 'City car',
    link: '/cars?f[0]=body:city-car&f[1]=vehicle_type:car',
    icon: CarCity,
  },
];
