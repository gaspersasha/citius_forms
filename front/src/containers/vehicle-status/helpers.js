export const steps = {
  SEARCH: {
    id: 'search',
    text: 'Search',
  },
  FINANCE: {
    id: 'finance',
    text: 'Finance',
  },
  ORDER: {
    id: 'order',
    text: 'Order',
  },
  VERIFY: {
    id: 'verify',
    text: 'Verify & confirm',
  },
  VEHICLE_PREP: {
    id: 'verify-prep',
    text: 'Vehicle prep',
  },
  DELIVERY: {
    id: 'delivery',
    text: 'Delivery',
  },
};

export const subSteps = {
  // search (finance flow)
  START: 'start',
  // finance (finance flow)
  SOFT: 'soft',
  APPLIED: 'applied',
  PART_EX: 'part-ex',

  // order (finance flow)
  PENDING_NO_PAID: 'pending-no-paid',
  PENDING_PAID: 'pending-paid',
  APPROVED_NO_PAID: 'approved-no-paid',
  APPROVED_PAID: 'approved-paid',
  SECCI_SENT: 'secci-sent', // common sub-step for finance and cash
  // order (cash flow)
  SECURE_VEHICLE: 'secure-vehicle',
  // verify (finance)
  GETTING_REQUIREMENTS: 'getting-requirements',
  LINK_SENT: 'link-sent',
  DOCS_SENT: 'docs-sent',
  CONDITION_REPORT: 'condition-report', // common sub-step for finance and cash
  // delivery (finance flow and cash flow - common)
  TO_BE_BOOKED: 'to-be-booked',
  BOOKED: 'booked',
  CONFIRMED: 'confirmed',
};

export const financeFlowSteps = [
  steps.SEARCH,
  steps.FINANCE,
  steps.ORDER,
  steps.VERIFY,
  steps.DELIVERY,
];
export const cashFlowSteps = [steps.ORDER, steps.VEHICLE_PREP, steps.DELIVERY];

export const getProgress = (subStep, isFinanceFlow) => {
  if (isFinanceFlow) {
    // finance flow
    switch (subStep) {
      // search
      case subSteps.START:
        return 50;
      // finance
      case subSteps.SOFT:
        return 25;
      case subSteps.APPLIED:
        return 50;
      case subSteps.PART_EX:
        return 75;
      // order
      case subSteps.PENDING_NO_PAID:
        return 17;
      case subSteps.PENDING_PAID:
        return 33;
      case subSteps.APPROVED_NO_PAID:
        return 50;
      case subSteps.APPROVED_PAID:
        return 67;
      case subSteps.SECCI_SENT:
        return 83;
      // verify
      case subSteps.GETTING_REQUIREMENTS:
        return 20;
      case subSteps.LINK_SENT:
        return 40;
      case subSteps.DOCS_SENT:
        return 60;
      case subSteps.CONDITION_REPORT:
        return 80;
      // delivery
      case subSteps.TO_BE_BOOKED:
        return 33;
      case subSteps.BOOKED:
        return 66;
      case subSteps.CONFIRMED:
        return 100;
      default:
        return 100;
    }
  }

  // cash flow
  switch (subStep) {
    // order
    case subSteps.SECURE_VEHICLE:
      return 33;
    case subSteps.SECCI_SENT:
      return 66;
    // vehicle prep
    case subSteps.CONDITION_REPORT:
      return 50;
    // delivery
    case subSteps.TO_BE_BOOKED:
      return 33;
    case subSteps.BOOKED:
      return 66;
    case subSteps.CONFIRMED:
      return 100;
    default:
      return 100;
  }
};
