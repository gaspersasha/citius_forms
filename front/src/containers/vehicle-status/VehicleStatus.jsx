import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { useRouter } from 'next/router';
import { OrderSummary, ProgressBarSteps, Spinner, Composer } from '~components';
import {
  OrderSummaryContextProvider,
  DeviceContextProvider,
  useDeviceContext,
} from '~contexts';
import { DEVICE_TYPES } from '~constants';
import { endpointFetch } from '~utils';
import VehiclePrep from '~containers/vehicle-status/VehiclePrep';
import { Search, Finance, Order, Verify, Delivery } from '.';
import { financeFlowSteps, cashFlowSteps, steps, getProgress } from './helpers';
import s from './styles/vehicle-status.module.sass';

const { DESKTOP, WIDE } = DEVICE_TYPES;

const getStepComponent = (
  step,
  subStep,
  quoteDetails,
  quoteId,
  quoteItemId,
  typoProps,
  isFinanceFlow
) => {
  const { vehicle, softCreditSearch, customerFinanceTerms, delivery } =
    quoteDetails;

  const { advertId, year, make, model, van } = vehicle || {};
  const { financeQuoteId } = customerFinanceTerms || {};
  const { bookedDate } = delivery || {};

  switch (step) {
    case steps.SEARCH.id:
      return (
        <Search
          productAdvertId={advertId}
          financeQuoteId={financeQuoteId}
          make={make}
          model={model}
          typoProps={typoProps}
        />
      );

    case steps.FINANCE.id:
      return (
        <Finance
          substep={subStep}
          productAdvertId={advertId}
          quoteItemId={quoteItemId}
          quoteId={quoteId}
          financeQuoteId={financeQuoteId}
          softCreditSearch={softCreditSearch || {}}
          typoProps={typoProps}
        />
      );

    case steps.ORDER.id:
      return (
        <Composer
          components={[OrderSummaryContextProvider, DeviceContextProvider]}
        >
          <Order
            substep={subStep}
            productAdvertId={advertId}
            financeQuoteId={financeQuoteId}
            isVan={van}
            typoProps={typoProps}
            isFinanceFlow={isFinanceFlow}
          />
        </Composer>
      );

    case steps.VERIFY.id:
      return <Verify substep={subStep} typoProps={typoProps} />;

    case steps.VEHICLE_PREP.id:
      return <VehiclePrep substep={subStep} typoProps={typoProps} />;

    case steps.DELIVERY.id:
      return (
        <Delivery
          substep={subStep}
          make={make}
          model={model}
          isVan={van}
          year={year}
          deliveryDate={bookedDate}
          typoProps={typoProps}
        />
      );

    default:
      return null;
  }
};

const VehicleStatus = () => {
  const [vehicleStatus, setVehicleStatus] = useState(null);
  const {
    query: { quoteId, productAdvertId, quoteItemId },
  } = useRouter();
  const { device } = useDeviceContext();

  const isDesktopMode = [DESKTOP, WIDE].includes(device);
  const align = isDesktopMode ? 'left' : 'center';

  const typoProps = {
    title: { align, type: 'h3', className: s.title },
    subTitle: { align, type: 'p', className: s.subTitle },
    text: { align, type: 'p', className: s.text },
  };

  useEffect(() => {
    if (!productAdvertId) return;
    endpointFetch('vehicleStatus', { productAdvertId })
      .then(setVehicleStatus)
      .catch(console.error);
  }, [productAdvertId]);

  if (!vehicleStatus) {
    return <Spinner />;
  }

  const {
    quoteDetails,
    quoteDetails: {
      step,
      subStep,
      quote: { paymentMethod },
    },
  } = vehicleStatus;
  const isFinanceFlow = paymentMethod === 'finance';

  return (
    <>
      <div className={s.wrapper}>
        <h1 className="page-title">My Account</h1>
        <div className={s.progressBar}>
          <ProgressBarSteps
            steps={isFinanceFlow ? financeFlowSteps : cashFlowSteps}
            currentStep={step}
            progress={getProgress(subStep, isFinanceFlow)}
          />
        </div>
        <div className={cn(s.container, 'form-container')}>
          <div className={s.subStep}>
            {getStepComponent(
              step,
              subStep,
              quoteDetails,
              quoteId,
              quoteItemId,
              typoProps,
              isFinanceFlow
            )}
          </div>
          <div className={s.orderSummaryWrapper}>
            <DeviceContextProvider>
              <OrderSummary data={vehicleStatus} />
            </DeviceContextProvider>
          </div>
        </div>
      </div>
      <DeviceContextProvider>
        <OrderSummary data={vehicleStatus} withExpandingWrapper />
      </DeviceContextProvider>
    </>
  );
};

const typoPropsFieldShape = PropTypes.shape({
  align: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
}).isRequired;

export const typoPropsShape = {
  title: typoPropsFieldShape,
  text: typoPropsFieldShape,
};

export default VehicleStatus;
