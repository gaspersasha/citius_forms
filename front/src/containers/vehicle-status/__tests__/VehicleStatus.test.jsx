import React from 'react';
import { DEVICE_TYPES } from '~constants';
import { endpointFetch } from '~utils';
import VehicleStatus, { Finance, Order, Verify, Delivery } from '..';
import { steps, subSteps } from '../helpers';

const { DESKTOP } = DEVICE_TYPES;

jest.mock('~utils', () => ({
  ...jest.requireActual('../../../utils'),
  endpointFetch: jest.fn(),
}));

global.mockUseRouter({
  useRouter: jest.fn(),
});

global.mockUseOrderSummaryContext({
  actions: {
    getDetails: jest.fn(),
  },
  state: {
    isOpen: false,
  },
});

global.mockUseDeviceContext({ device: DESKTOP });

const quoteDetails = {
  step: steps.SEARCH.id,
  subStep: subSteps.START,
  vehicle: {
    advertId: 1556200,
    make: 'vauxhall',
    model: 'corsa hatchback special eds',
    van: true,
    year: 2010,
  },
  customerFinanceTerms: {
    financeQuoteId: 3735215,
  },
  lenderFinanceTerms: {},
  quote: {},
  partex: [],
  financeApplication: {},
  softCreditSearch: {
    band: 'B',
    score: 420,
  },
  reservation: {},
  order: {},
  verification: {},
  delivery: { bookedDate: 'Tue Sep 16 13:33:50 BST 2021' },
};

global.mockUseRouter({
  pathname: '/account/vehicle-status',
  query: {
    quoteId: 10500361,
    quoteItemId: 1523212,
    productAdvertId: 1020046,
  },
});

describe('<VehicleStatus />', () => {
  it('should match snapshot (search)', async () => {
    endpointFetch.mockResolvedValueOnce({ quoteDetails });
    const wrapper = mount(<VehicleStatus />);

    await global.updateWrapper(wrapper);

    expect(wrapper).toMatchSnapshot();
  });

  it('handles finance step case)', async () => {
    endpointFetch.mockResolvedValueOnce({
      quoteDetails: {
        ...quoteDetails,
        step: steps.FINANCE.id,
        subStep: subSteps.SOFT,
      },
    });
    const wrapper = mount(<VehicleStatus />);

    await global.updateWrapper(wrapper);

    expect(wrapper.find(Finance).exists()).toBe(true);
  });

  it('handles order step case', async () => {
    endpointFetch.mockResolvedValueOnce({
      quoteDetails: {
        ...quoteDetails,
        step: steps.ORDER.id,
        subStep: subSteps.PENDING_NO_PAID,
      },
    });
    const wrapper = mount(<VehicleStatus />);

    await global.updateWrapper(wrapper);

    expect(wrapper.find(Order).exists()).toBe(true);
  });

  it('handles verify step case', async () => {
    endpointFetch.mockResolvedValueOnce({
      quoteDetails: {
        ...quoteDetails,
        step: steps.VERIFY.id,
        subStep: subSteps.LINK_SENT,
      },
    });
    const wrapper = mount(<VehicleStatus />);

    await global.updateWrapper(wrapper);

    expect(wrapper.find(Verify).exists()).toBe(true);
  });

  it('handles delivery step case', async () => {
    endpointFetch.mockResolvedValueOnce({
      quoteDetails: {
        ...quoteDetails,
        step: steps.DELIVERY.id,
        subStep: subSteps.BOOKED,
      },
    });
    const wrapper = mount(<VehicleStatus />);

    await global.updateWrapper(wrapper);

    expect(wrapper.find(Delivery).exists()).toBe(true);
  });
});
