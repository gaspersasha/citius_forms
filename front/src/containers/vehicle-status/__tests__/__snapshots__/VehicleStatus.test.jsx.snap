// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VehicleStatus /> should match snapshot (search) 1`] = `
<VehicleStatus>
  <div
    className="wrapper"
  >
    <h1
      className="page-title"
    >
      My Account
    </h1>
    <div
      className="progressBar"
    >
      <ProgressBarSteps
        currentStep="search"
        progress={100}
        steps={
          Array [
            Object {
              "id": "order",
              "text": "Order",
            },
            Object {
              "id": "verify-prep",
              "text": "Vehicle prep",
            },
            Object {
              "id": "delivery",
              "text": "Delivery",
            },
          ]
        }
      >
        <div
          className="container"
        >
          <div
            className="bar"
          >
            <div
              className="step future"
              data-id="progress-bar-step"
              key="order"
            >
              <div
                className="icon"
              >
                <span
                  className="stepNumber"
                >
                  1
                </span>
              </div>
              <div
                className="title"
              >
                Order
              </div>
            </div>
            <div
              className="step future"
              data-id="progress-bar-step"
              key="verify-prep"
            >
              <div
                className="icon"
              >
                <span
                  className="stepNumber"
                >
                  2
                </span>
              </div>
              <div
                className="title"
              >
                Vehicle prep
              </div>
            </div>
            <div
              className="step future"
              data-id="progress-bar-step"
              key="delivery"
            >
              <div
                className="icon"
              >
                <span
                  className="stepNumber"
                >
                  3
                </span>
              </div>
              <div
                className="title"
              >
                Delivery
              </div>
            </div>
          </div>
        </div>
      </ProgressBarSteps>
    </div>
    <div
      className="container form-container"
    >
      <div
        className="subStep"
      >
        <Search
          financeQuoteId={3735215}
          make="vauxhall"
          model="corsa hatchback special eds"
          productAdvertId={1556200}
          typoProps={
            Object {
              "subTitle": Object {
                "align": "left",
                "className": "subTitle",
                "type": "p",
              },
              "text": Object {
                "align": "left",
                "className": "text",
                "type": "p",
              },
              "title": Object {
                "align": "left",
                "className": "title",
                "type": "h3",
              },
            }
          }
        >
          <p
            className="pretitle"
          >
            Let's get started!
          </p>
          <Typography
            align="left"
            className="title subStepTitleMargin"
            color="dark"
            type="h3"
          >
            <h3
              className="typographyComponent txt-dark txt-left title subStepTitleMargin"
            >
              How do you intend to purchase your
               Vauxhall Corsa hatchback special eds
              ?
            </h3>
          </Typography>
          <div
            className="block"
          >
            <Typography
              align="center"
              className="title"
              color="dark"
              type="h3"
            >
              <h3
                className="typographyComponent txt-dark txt-center title"
              >
                I want to purchase with finance
              </h3>
            </Typography>
            <Typography
              align="center"
              className="text"
              color="dark"
              type="p"
            >
              <p
                className="typographyComponent txt-dark txt-center text"
              >
                Pay for the car in monthly installments with a finance deal from one of the lenders on our panel.
              </p>
            </Typography>
            <Typography
              align="center"
              className="text"
              color="dark"
              type="p"
            >
              <p
                className="typographyComponent txt-dark txt-center text"
              >
                Find out more about 
                <a
                  className="link"
                  href="/car-finance/198/buying-a-car-on-finance"
                  target="_blank"
                >
                  car finance
                </a>
              </p>
            </Typography>
            <Button
              className="btn"
              disabled={false}
              href="/finance-eligibility?productAdvertId=1556200&financeQuoteId=3735215"
              onClick={null}
              selected={false}
              styleType="primary"
              target="_blank"
              type="submit"
            >
              <a
                className="buttonComponent primary btn"
                disabled={false}
                href="/finance-eligibility?productAdvertId=1556200&financeQuoteId=3735215"
                onClick={null}
                target="_blank"
                type="submit"
              >
                Pay monthly
              </a>
            </Button>
          </div>
          <div
            className="block"
          >
            <Typography
              align="center"
              className="title"
              color="dark"
              type="h3"
            >
              <h3
                className="typographyComponent txt-dark txt-center title"
              >
                I want to buy outright
              </h3>
            </Typography>
            <Typography
              align="center"
              className="text"
              color="dark"
              type="p"
            >
              <p
                className="typographyComponent txt-dark txt-center text"
              >
                Pay for the car upfront in full.
                <br />
                <br />
                To begin, reserve your car for just
                 £199 
                and we'll get to work securing it for you. You will pay the remaining balance before delivery.
              </p>
            </Typography>
            <Button
              className="btn"
              disabled={false}
              href="/reservation?productAdvertId=1556200&financeQuoteId=3735215"
              onClick={null}
              selected={false}
              styleType="primary"
              target="_blank"
            >
              <a
                className="buttonComponent primary btn"
                disabled={false}
                href="/reservation?productAdvertId=1556200&financeQuoteId=3735215"
                onClick={null}
                target="_blank"
              >
                Buy outright
              </a>
            </Button>
          </div>
        </Search>
      </div>
      <div
        className="orderSummaryWrapper"
      >
        <DeviceContextProvider>
          <OrderSummary
            WITH_PX_CTA={true}
            carDetails={
              Object {
                "name": null,
                "price": null,
                "year": null,
              }
            }
            data={
              Object {
                "quoteDetails": Object {
                  "customerFinanceTerms": Object {
                    "financeQuoteId": 3735215,
                  },
                  "delivery": Object {
                    "bookedDate": "Tue Sep 16 13:33:50 BST 2021",
                  },
                  "financeApplication": Object {},
                  "lenderFinanceTerms": Object {},
                  "order": Object {},
                  "partex": Array [],
                  "quote": Object {},
                  "reservation": Object {},
                  "softCreditSearch": Object {
                    "band": "B",
                    "score": 420,
                  },
                  "step": "search",
                  "subStep": "start",
                  "vehicle": Object {
                    "advertId": 1556200,
                    "make": "vauxhall",
                    "model": "corsa hatchback special eds",
                    "van": true,
                    "year": 2010,
                  },
                  "verification": Object {},
                },
              }
            }
            withExpandingWrapper={false}
          >
            <Contents
              WITH_PX_CTA={true}
              carDetails={
                Object {
                  "name": null,
                  "price": null,
                  "year": null,
                }
              }
              className="order-summary"
              data={
                Object {
                  "quoteDetails": Object {
                    "customerFinanceTerms": Object {
                      "financeQuoteId": 3735215,
                    },
                    "delivery": Object {
                      "bookedDate": "Tue Sep 16 13:33:50 BST 2021",
                    },
                    "financeApplication": Object {},
                    "lenderFinanceTerms": Object {},
                    "order": Object {},
                    "partex": Array [],
                    "quote": Object {},
                    "reservation": Object {},
                    "softCreditSearch": Object {
                      "band": "B",
                      "score": 420,
                    },
                    "step": "search",
                    "subStep": "start",
                    "vehicle": Object {
                      "advertId": 1556200,
                      "make": "vauxhall",
                      "model": "corsa hatchback special eds",
                      "van": true,
                      "year": 2010,
                    },
                    "verification": Object {},
                  },
                }
              }
              isDesktopOnly={true}
            >
              <div
                className="flSummary desktopOnly order-summary"
              >
                <div
                  className="mainTitle"
                >
                  Order summary
                </div>
                <div
                  className="vehicleDetailsBlock"
                >
                  <img
                    alt="full-car"
                    className="image"
                    src="https://d13fze08x8d2oj.cloudfront.netundefined"
                  />
                  <div
                    className="vehicleDetails"
                  >
                    <span
                      className="make"
                    >
                      vauxhall corsa hatchback special eds
                    </span>
                    <span
                      className="year"
                    >
                      2010
                    </span>
                    <span
                      className="cashPrice"
                    >
                      cash price £0
                    </span>
                  </div>
                </div>
                <a
                  className="advert"
                  href="deal-1020046"
                  rel="noreferrer"
                  target="_blank"
                >
                  View advert
                </a>
                <div
                  className="separator"
                />
                <div
                  className="partExchange"
                  data-id="order-summary-part-exchange"
                >
                  <div
                    className="addPartExchange"
                  >
                    <span>
                      Add part exchange?
                    </span>
                    <span>
                      See what you can get from our partners The Car Buying Group
                    </span>
                    <Button
                      className=""
                      disabled={false}
                      href="part-exchange"
                      onClick={null}
                      selected={false}
                      styleType="primary"
                    >
                      <a
                        className="buttonComponent primary"
                        disabled={false}
                        href="part-exchange"
                        onClick={null}
                      >
                        Add part exchange
                      </a>
                    </Button>
                  </div>
                </div>
              </div>
            </Contents>
          </OrderSummary>
        </DeviceContextProvider>
      </div>
    </div>
  </div>
  <DeviceContextProvider>
    <OrderSummary
      WITH_PX_CTA={true}
      carDetails={
        Object {
          "name": null,
          "price": null,
          "year": null,
        }
      }
      data={
        Object {
          "quoteDetails": Object {
            "customerFinanceTerms": Object {
              "financeQuoteId": 3735215,
            },
            "delivery": Object {
              "bookedDate": "Tue Sep 16 13:33:50 BST 2021",
            },
            "financeApplication": Object {},
            "lenderFinanceTerms": Object {},
            "order": Object {},
            "partex": Array [],
            "quote": Object {},
            "reservation": Object {},
            "softCreditSearch": Object {
              "band": "B",
              "score": 420,
            },
            "step": "search",
            "subStep": "start",
            "vehicle": Object {
              "advertId": 1556200,
              "make": "vauxhall",
              "model": "corsa hatchback special eds",
              "van": true,
              "year": 2010,
            },
            "verification": Object {},
          },
        }
      }
      withExpandingWrapper={true}
    />
  </DeviceContextProvider>
</VehicleStatus>
`;
